portfolio/src/main/java/com/google/sps/servlets/CommentServlet.java

```diff
@@ -26,56 +24,95 @@
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
-
+import java.time.Instant;
 import java.time.LocalDateTime;
 import java.util.ArrayList;
 import java.util.List;
+import java.util.TimeZone;
 
 /** Servlet that handles getting and posting comment content. */
 @WebServlet("/comment")
 public final class CommentServlet extends HttpServlet {
-  private final List<Comment> comments = new ArrayList<>();
   
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-     String json = convertToJsonUsingGson(this.comments);
+    // Loads the comment from Datastore
+    List<Comment> comments = getComments();
 
-    // Send the JSON as the response
-    response.setContentType("application/json;");
-    response.getWriter().println(json);
+    // Converts into json form
+    String json = convertToJsonUsingGson(comments);
+
+    // Sends the JSON as the response
+    sendJsonResponse(response, json);
   }
 
+  /** Loads the comment from Datastore */
+  private List<Comment> getComments() {
+    Query query = new Query("Comment").addSort("timestamp", Query.SortDirection.DESCENDING);
+
+    DatastoreService datastoreService = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastoreService.prepare(query);
+
+    List<Comment> comments = new ArrayList<>();
+    for (Entity entity: results.asIterable()) {
+      long id = entity.getKey().getId();
+      String commenter = (String) entity.getProperty("name");
```

Comment:
Conversion between Entity and Comment are now scattered in various places with hardcoded tag name. If there're some changes in those name, hard to trace bugs may arise. Maybe we can add toEntity and fromEntity method into Comment class? This way we have a symmetric pair of conversion between the two?

Reply:
Fixed